traffic_ops/traffic_ops_golang/cachegroup/dspost.go

```diff
@@ -0,0 +1,290 @@
+	if err := updateDSParam(tx, dsIDs, "hdr_rw_", "edge_header_rewrite"); err != nil {
+		return err
+	}
+	if err := updateDSParam(tx, dsIDs, "cacheurl_", "cacheurl"); err != nil {
+		return err
+	}
+	if err := updateDSParam(tx, dsIDs, "regex_remap_", "regex_remap"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func updateDSParam(tx *sql.Tx, dsIDs []int64, paramPrefix string, dsField string) error {
+	_, err := tx.Exec(`
+DELETE FROM parameter
+WHERE name = 'location'
+AND config_file IN (
+  SELECT CONCAT('`+paramPrefix+`', xml_id, '.config')
+  FROM deliveryservice as ds
+  WHERE ds.id = ANY($1)
+  AND (ds.`+dsField+` IS NULL OR ds.`+dsField+` = '')
+)
+`, pq.Array(dsIDs))
+	if err != nil {
+		return err
+	}
+
+	rows, err := tx.Query(`
+WITH ats_config_location AS (
+  SELECT TRIM(TRAILING '/' FROM value) as v FROM parameter WHERE name = 'location' AND config_file = 'remap.config'
+)
+INSERT INTO parameter (name, config_file, value) (
+  SELECT
+    'location' as name,
+    CONCAT('`+paramPrefix+`', xml_id, '.config'),
+    (select v from ats_config_location)
+  FROM deliveryservice WHERE id = ANY($1)
+) ON CONFLICT (name, config_file, value) DO UPDATE SET id = EXCLUDED.id RETURNING id
+`, pq.Array(dsIDs))
+	if err != nil {
+		return errors.New("inserting parameters: " + err.Error())
+	}
+	ids := []int64{}
+	for rows.Next() {
+		id := int64(0)
+		if err := rows.Scan(&id); err != nil {
+			return errors.New("scanning inserted parameters: " + err.Error())
+		}
+		ids = append(ids, id)
+	}
+
+	_, err = tx.Exec(`
+INSERT INTO profile_parameter (parameter, profile) (
+  SELECT UNNEST($1::int[]), server.profile
+  FROM server
+  JOIN deliveryservice_server as dss ON dss.server = server.id
+  JOIN deliveryservice as ds ON ds.id = dss.deliveryservice
+  WHERE ds.id = ANY($2)
+) ON CONFLICT DO NOTHING
+`, pq.Array(ids), pq.Array(dsIDs))
+	if err != nil {
+		return errors.New("inserting profile parameters: " + err.Error())
+	}
+	return nil
+}
+
+func deleteConfigFile(tx *sql.Tx, configFile string) error {
+	_, err := tx.Exec(`DELETE FROM parameter WHERE name = 'location' AND config_file = $1`, configFile)
+	return err
+}
+
+func getDSTenants(tx *sql.Tx, dsIDs []int64) ([]int64, error) {
+	q := `
+SELECT COALESCE(tenant_id, 0) FROM deliveryservice
+WHERE deliveryservice.id = ANY($1)
+`
+	rows, err := tx.Query(q, pq.Array(dsIDs))
+	if err != nil {
+		return nil, errors.New("selecting delivery service tenants: " + err.Error())
+	}
+	defer rows.Close()
+	tenantIDs := []int64{}
+	for rows.Next() {
+		id := int64(0)
+		if err := rows.Scan(&id); err != nil {
+			return nil, errors.New("querying cachegroup delivery service tenants: " + err.Error())
+		}
+		tenantIDs = append(tenantIDs, id)
+	}
+	return tenantIDs, nil
+}
+
+// FinishTx commits the transaction if commit is true when it's called, otherwise it rolls back the transaction. This is designed to be called in a defer.
+func FinishTx(tx *sql.Tx, commit *bool) {
```

Comment:
Is this something that needs to be implemented on all Handlers now?

Reply:
No, this was a legacy thing, because this PR was made before everything was changed to use transactions. I removed it.